### ISSUE TITLE
SingleQuery regression

### ISSUE THREAD

**Issue by sebastienros** [aspnet/Benchmarks#1923]
![image](https://github.com/aspnet/Benchmarks/assets/1165805/f2699b7e-16df-48d5-a4af-0dfbbb001cdd)

**Comment by sebastienros** [comment]
Nothing obvious from aspnetcore and runtime. 

https://github.com/dotnet/aspnetcore/compare/faf594a06685...e3aba084cde2
https://github.com/dotnet/runtime/compare/60b77a63df30...567f81e71909

But matches the npgsql upgrade to rc2. Haven't confirmed it with a custom run.

**Comment by sebastienros** [comment]
@roji

**Comment by roji** [comment]
This is pretty much expected... As @NinoFloris wrote elsewhere we'll probably improve a bit but not right away.

/cc @NinoFloris @vonzshik

**Comment by NinoFloris** [comment]
I'm confident we'll be able to get back to baseline once we have some time to do runs on the perflab. 

We have done some runs a while back of this PR https://github.com/npgsql/npgsql/pull/5245 which showed a decent improvement in fortunes, some of that is bound to translate to single query as well.

**Comment by NinoFloris** [comment]
OK after merging https://github.com/npgsql/npgsql/pull/5245#issuecomment-1791271350 we improved fortunes by going from 479,772 to 497,363 RPS, so about a ~4% improvement (pr contains all run details).

Running single query on the same pr got us back up to ~510k (before: 499,078, after: 508,344). However with the average before the regression being ~530k we were still missing some perf.

I just merged https://github.com/npgsql/npgsql/pull/5371#issuecomment-1792181934 to get us all the way there. Going from 514,077 (first perf pr) to 531,385; matching the previous average. This PR might also improve fortunes by another 0.5%, we'll have to watch the trend a bit more to be sure.

In summary we've fixed the single query regression and improved fortunes RPS by 4%.

@sebastienros both https://github.com/npgsql/npgsql/pull/5245 and https://github.com/npgsql/npgsql/pull/5371 are merged so this can be closed, thanks!
